index.js

```diff
@@ -313,4 +386,95 @@ function reducer(acc, cur) {
   return acc;
 }
 
+/**
+ * Given an error from got, see if Shopify told us how long to wait before
+ * retrying. Return a duration in ms if we can, and otherwise return null.
+ *
+ * @param {Object} error Error object from got call
+ * @return {Boolean | null}
+ * @private
+ **/
+function maybeRetryMS(error) {
+  // for simplicity, retry network connectivity issues after a hardcoded 1s
+  if (
+    typeof error === 'object' &&
+    'code' in error &&
+    RetryableErrorCodes.has(error.code)
+  ) {
+    return 1000;
+  }
+
+  const response = error.response;
+  if (!response || !response.headers) {
+    return null;
+  }
+
+  if (response.headers && response.headers['retry-after']) {
+    const value = parseFloat(response.headers['retry-after']);
+    if (isNaN(value)) {
+      const when = new Date(value).valueOf();
+      return when - Date.now();
+    } else if (isFinite(value)) {
+      return value;
```

Comment:
```suggestion
      return value * 1000;
```

Wouldn't it be simpler to remove `parseFloat` and all the check and just return `response.headers['retry-after'] * 1000 || null`?